clippy_lints/src/methods/option_filter_map.rs

```diff
@@ -0,0 +1,21 @@
+use crate::utils::{match_qpath, paths};
+use rustc_hir::{self, Expr, ExprKind};
+use std::iter;
+
+fn calls_option_method<'tcx>(expr: &'tcx [Expr<'_>], fn_name: &str) -> bool {
+    expr.get(1).map_or(false, |x| match &x.kind {
+        ExprKind::Path(qp) => match_qpath(
+            qp,
+            &paths::OPTION
+                .iter()
+                .cloned()
+                .chain(iter::once(fn_name))
+                .collect::<Vec<&str>>(),
+        ),
+        _ => false,
+    })
+}
+
+pub fn in_scope<'tcx>(filter_args: &'tcx [Expr<'_>], map_args: &'tcx [Expr<'_>]) -> bool {
```

Comment:
This doesn't have to be in its own module. And NIT: the `in_scope` function name doesn't really make sense to me. Maybe move it to `mod.rs` and call it `is_option_filter_map`.